hearth/lib/hearth/http/response.rb

```diff
@@ -39,7 +39,7 @@ def reset
         @status = 0
         @reason = nil
         @fields.clear
-        @body.truncate(0)
+        @body.truncate(0) if @body.respond_to?(:truncate)
```

Comment:
Not sure about this one. Previous logic was checking truncate to determine if something is retryable. Now it's just always retryable, and we don't truncate if the body doesn't allow for it. But does this break?

Reply:
Undo this